sample/kotlin/src/main/kotlin/com/datadog/android/sample/widget/DatadogWidgetsProvider.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sample.widget
+
+import android.app.PendingIntent
+import android.appwidget.AppWidgetManager
+import android.appwidget.AppWidgetProvider
+import android.content.Context
+import android.content.Intent
+import android.widget.RemoteViews
+import com.datadog.android.sample.R
+
+/**
+ * Implementation of App Widget functionality.
+ */
+class DatadogWidgetsProvider : AppWidgetProvider() {
+    override fun onUpdate(
+        context: Context,
+        appWidgetManager: AppWidgetManager,
+        appWidgetIds: IntArray
+    ) {
+        // There may be multiple widgets active, so update all of them
+        for (appWidgetId in appWidgetIds) {
+            updateAppWidget(
+                context,
+                appWidgetManager,
+                appWidgetId
+            )
+        }
+    }
+
+    override fun onEnabled(context: Context) {
+        // Enter relevant functionality for when the first widget is created
+    }
+
+    override fun onDisabled(context: Context) {
+        // Enter relevant functionality for when the last widget is disabled
+    }
+
+    companion object {
+
+        internal fun updateAppWidget(
```

Comment:
That method is a bit long, maybe we can split it into smaller chunks ?

Reply:
Also why do you need to make it a static method? It seems really weird :/

Reply:
That's the way they propose...it was actually auto - generated...not sure why 🤔

Reply:
I have actually the explanation...the way the widget works is that whenever it gets an update (by clicking on a button or something) it will recreate a new Receiver or IntentService depending where you are handling this action (what PendingIntent you are using). I think the reason why they want to have static methods there is to keep that object as light as possible in case is recreated multiple times. Apart this I don't really have any other explanation.